awx/api/serializers.py

```diff
@@ -3406,9 +3412,15 @@ def get_related(self, obj):
 class WorkflowApprovalListSerializer(WorkflowApprovalSerializer, UnifiedJobListSerializer):
 
     can_approve_or_deny = serializers.SerializerMethodField()
+    approval_expiration = serializers.SerializerMethodField()
 
     class Meta:
-        fields = ('*', '-execution_node', '-controller_node', 'can_approve_or_deny')
+        fields = ('*', '-execution_node', '-controller_node', 'can_approve_or_deny', 'approval_expiration')
+
+    def get_approval_expiration(self, obj):
+        if obj.status != 'pending' or obj.timeout == 0:
+            return None
+        return now() + timedelta(seconds=obj.timeout)
```

Comment:
Same comment here.

Reply:
This method shouldn't be duplicated here. Defining it once on `WorkflowApprovalSerializer` should be enough. If there's concern about the implementation, then we should have a test that checks its presence in list vs. detail views.

Reply:
Actually @AlanCoding given that this is a subclass of `WorkflowApprovalSerializer`, it seems like _most_ of this is duplicated, and probably unnecessary.  Shouldn't we just need this?

```python
class WorkflowApprovalListSerializer(WorkflowApprovalSerializer, UnifiedJobListSerializer):
    pass
```

Reply:
if that was true, then we could delete `WorkflowApprovalListSerializer` entirely... unless there's something that `UnifiedJobListSerializer` is doing that we want. If we do want something from that (I do not know what), then I agree with what you're saying.